DHCPServer/Library/DHCPServer.cs

```diff
@@ -22,21 +22,47 @@ THE SOFTWARE.
 
 */
 using System;
-using System.Collections;
 using System.Collections.Generic;
 using System.IO;
 using System.Net;
-using System.Text;
-using System.Net.Sockets;
-using System.Net.NetworkInformation;
-using System.Net.Configuration;
 using System.Threading;
 using System.Linq;
+using Library;
+using DHCPServerApp;
 
 namespace CodePlex.JPMikkers.DHCP
 {
     public class DHCPServer : IDHCPServer
     {
+        public static DHCPServer FromConfig(DHCPServerConfiguration m_Config)

```

Comment:
This explicitly ties the DHCPServer class to the xml configuration stuff, please move this method to a static Extension\DHCPServerFactory.cs class